src/in-depth/signals.md

```diff
@@ -1 +1,134 @@
 # Signal handling
+
+Processes
+like command line applications
+need to react to signals sent by the operating system.
+The most common example is probably <kbd>Ctrl</kbd>+<kbd>C</kbd>,
+the signal that typically tells a process to terminate.
+To handle signals in Rust programs
+you need to consider how you can receive these signals
+as well as how you can react to them.
+
+<aside>
+
+**Note:**
+If your applications does not require special cleanup,
+the default handling is fine
+(i.e. exit immediately and let the OS cleanup).
+In that case:
+No need to do what this chapter tells you!
+For applications that need to clean up after themselves,
+for example to close network connections,
+remove temporary files,
+or reset system settings,
+this chapter is however very relevant!
```

Comment:
The other chapter probably sounds better (and it probably is not specific to CLI only). Still, as a wording nitpick, maybe not call it „cleanup“, but „graceful shutdown“? Like, the OS will terminate network connections, but it might happen in the middle of a message instead of after saying proper „Good bye“.

Reply:
Thanks! Expanded the note to describe graceful shutdown scenarios, and opened #107 for tracking ideas around a "robustness" chapter.